shopfloor_base/models/shopfloor_app.py

```diff
@@ -0,0 +1,262 @@
+    """Backend for a Shopfloor app."""
+
+    _name = "shopfloor.app"
+    _inherit = "collection.base"
+    _description = "A Shopfloor application"
+
+    name = fields.Char(required=True, translate=True)
+    short_name = fields.Char(
+        required=True, translate=True, help="Needed for app manifest"
+    )
+    # Unique name
+    tech_name = fields.Char(required=True, index=True)
+    active = fields.Boolean(default=True)
+    category = fields.Selection(selection=[("", "None")])
+    api_route = fields.Char(
+        compute="_compute_api_route",
+        compute_sudo=True,
+        help="Base route for endpoints attached to this app, public version.",
+    )
+    api_route = fields.Char(
+        compute="_compute_api_route",
+        compute_sudo=True,
+        help="""
+        Base route for endpoints attached to this app,
+        internal controller-ready version.
+        """,
+    )
+    url = fields.Char(compute="_compute_url", help="Public URL to use the app.")
+    api_docs_url = fields.Char(compute="_compute_url", help="Public URL for api docs.")
+    auth_type = fields.Selection(
+        selection="_selection_auth_type", default="user_endpoint"
+    )
+    registered_routes = fields.Text(
+        compute="_compute_registered_routes",
+        compute_sudo=True,
+        help="Technical field to allow developers to check registered routes on the form",
+        groups="base.group_no_one",
+    )
+    profile_ids = fields.Many2many(
+        comodel_name="shopfloor.profile",
+        string="Profiles",
+        help="Profiles used by this app. "
+        "This will determine menu items too."
+        "However this field is not required "
+        "in case you don't need profiles and menu items from the backend.",
+    )
+    profile_required = fields.Boolean(compute="_compute_profile_required", store=True)
+
+    _sql_constraints = [("tech_name", "unique(tech_name)", "tech_name must be unique")]
+
+    _api_route_path = "/shopfloor/api/"
+
+    @api.depends("tech_name")
+    def _compute_api_route(self):
+        for rec in self:
+            rec.api_route = rec._api_route_path + rec.tech_name
+
+    _base_url_path = "/shopfloor/app/"
+    _base_api_docs_url_path = "/shopfloor/api-docs/"
+
+    @api.depends("tech_name")
+    def _compute_url(self):
+        for rec in self:
+            rec.url = rec._base_url_path + rec.tech_name
+            rec.api_docs_url = rec._base_api_docs_url_path + rec.tech_name
+
+    @api.depends("tech_name")
+    def _compute_registered_routes(self):
+        for rec in self:
+            routes = rec._registered_routes()
+            rec.registered_routes = "\n".join(sorted([x[1].rule for x in routes]))
+
+    @api.depends("profile_ids")
+    def _compute_profile_required(self):
+        for rec in self:
+            rec.profile_required = bool(rec.profile_ids)
+
+    def _selection_auth_type(self):
+        return self.env["endpoint.route.handler"]._selection_auth_type()
```

Comment:
user_endoint needs the module shopfloor_mobile_base_auth_user
Either it should then be in the dependencies or that selection option should be added by that module.

Reply:
Even with the module installed, I still get:
[Vue warn]: Error in created hook: "Error: Auth type 'user_endpoint' not supported"

Reply:
I think this is solved now.